crates/ruff_linter/src/rules/airflow/rules/deprecated_members.rs

```diff
@@ -0,0 +1,92 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::Expr;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+#[derive(Debug, Eq, PartialEq)]
+enum Replacement {
+    None,
+    Name(String),
+}
+
+impl Replacement {
+    fn name(name: impl Into<String>) -> Self {
+        Self::Name(name.into())
+    }
+}
+
+/// ## What it does
+/// Checks for uses of deprecated Airflow functions and values.
+///
+/// ## Why is this bad?
+/// Airflow 3.0 removed various deprecated functions, members, and other
+/// values. Some have more modern replacements. Others are considered too niche
+/// and not worth to be maintained in Airflow.
+///
+/// ## Example
+/// ```python
+/// from airflow.utils.dates import days_ago
+///
+///
+/// yesterday = days_ago(today, 1)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// from datetime import timedelta
+///
+///
+/// yesterday = today - timedelta(days=1)
+/// ```
```

Comment:
Is there some documentation available already, such as a migration guide?

Reply:
Unfortunately there’s not. We may create it later, but would it also make sense to use this as the documentation? I can make the description here a bit longer, and the guide can link back to these rules instead.

Reply:
It probably "depends," but I have a slight preference toward having this documentation outside of Ruff, at least if it should also cover *why* something has been deprecated and provide guidance with what it should be replaced instead. Otherwise, the "burden" for maintaining the list falls on us and I don't see us having the authority or knowledge to maintain the list.

Reply:
Got it, that makes sense. We will create a separate migration guide then. This will be done later and we’ll add the link here.